### PR TITLE
added authorizer support for collect-reward

### PR DESCRIPTION
**Pull request by Zorro30**
A brief description of the changes in this PR:

Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/zwalletcli/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- 0chain:
- Other: ...

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/0chain/zwalletcli
 go 1.20
 
 require (
-	github.com/0chain/gosdk v1.11.7
+	github.com/0chain/gosdk v1.11.4-0.20240118170536-41dc66bf45b7
```
Comment (Zorro30) [inline go.mod]: Added older version just to test!

I'll update the tag once the go-sdk code is merged!

**Review by dabasov — APPROVED** [review]
lgtm

File: go.mod (hunk 1 above)
Comment (Zorro30) [reply go.mod]: Updated back to latest version

**Review by dabasov — APPROVED** [review]
lgtm

**Review by dabasov — COMMENTED** [review]
lgtm

**Pull request merged**